zokrates_core/src/flatten/mod.rs

```diff
@@ -464,9 +464,9 @@ impl<'ast, T: Field> Flattener<'ast, T> {
                 }
             },
             e => {
-                // we have array[e] with e an arbitrary expression
+                // we have array[e] with e as an arbitrary expression
```

Comment:
I don't agree with this change. The intent is similar as "f(x) with x an integer"

Reply:
Fixed in 3399c92